NRAS.py

```diff
@@ -0,0 +1,101 @@
+from abc import ABC, abstractmethod
+
+
+class BiologicalSequence(ABC):
+
+    @abstractmethod
+    def __getitem__(self, index) -> str:
+        pass
+
+    @abstractmethod
+    def __str__(self) -> str:
+        pass
+
+    @abstractmethod
+    def __repr__(self) -> str:
+        pass
+
+    @abstractmethod
+    def check_alphabet(self) -> bool:
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    def __init__(self, seq: str):
+        self.seq = seq
+        self.length = len(self.seq)
+        self.check_alphabet()
+
+    def __getitem__(self, slc) -> str:
+        if isinstance(slc, int):
+            return self.seq.__getitem__(slc - 1)
+        elif isinstance(slc, slice):
+            new_slice = slice(slc.start - 1, slc.stop, slc.step)
+            return self.seq.__getitem__(new_slice)
+
+    def __str__(self) -> str:
+        return f'{self.seq}'
+
+    def __repr__(self) -> str:
+        return f'{self.seq}'
+
+    def check_alphabet(self) -> bool:
+        unique_char_seq = set(self.seq)
+        return all(nuc in self.complement_dict for nuc in unique_char_seq)
```

Comment:
Если в дальнейшем используешь в качестве проверки на ините, неплохо было бы поднимать ошибку, а не возвращать значение
```suggestion
        if not all(nuc in self.complement_dict for nuc in unique_char_seq):
            raise ValueError('Sequence has wrong characters!')
```